tests/integration-tests/tests/update/test_update.py

```diff
@@ -749,3 +761,295 @@ def _test_update_queue_strategy_with_running_job(
     _check_queue_ami(cluster, ec2, pcluster_ami_id, "queue1")
     _check_queue_ami(cluster, ec2, pcluster_copy_ami_id, "queue2")
     assert_compute_node_states(scheduler_commands, queue1_nodes, "idle")
+
+
+@pytest.mark.usefixtures("instance")
+def test_dynamic_file_systems_update(
+    region,
+    os,
+    pcluster_config_reader,
+    ami_copy,
+    clusters_factory,
+    scheduler_commands_factory,
+    request,
+    snapshots_factory,
+    efs_stack_factory,
+    vpc_stack,
+    key_name,
+    s3_bucket_factory,
+    test_datadir,
+    fsx_factory,
+    svm_factory,
+    open_zfs_volume_factory,
+):
+    """Test update shared storages."""
+    existing_ebs_mount_dir = "/existing_ebs_mount_dir"
+    existing_efs_mount_dir = "/existing_efs_mount_dir"
+    fsx_lustre_mount_dir = "/fsx_lustre_mount_dir"
+    fsx_ontap_mount_dir = "/fsx_ontap_mount_dir"
+    fsx_open_zfs_mount_dir = "/fsx_open_zfs_mount_dir"
+    new_ebs_mount_dir = "/new_ebs_mount_dir"
+    new_raid_mount_dir = "/raid_dir"
+    new_efs_mount_dir = "/new_efs_mount_dir"
+    new_lustre_mount_dir = "/new_lustre_mount_dir"
+    ebs_mount_dirs = [new_ebs_mount_dir, existing_ebs_mount_dir]
+    efs_mount_dirs = [existing_efs_mount_dir, new_efs_mount_dir]
+    fsx_mount_dirs = [new_lustre_mount_dir, fsx_lustre_mount_dir, fsx_open_zfs_mount_dir, fsx_ontap_mount_dir]
+    all_mount_dirs = ebs_mount_dirs + [new_raid_mount_dir] + efs_mount_dirs + fsx_mount_dirs
+
+    bucket_name = s3_bucket_factory()
+    bucket = boto3.resource("s3", region_name=region).Bucket(bucket_name)
+    bucket.upload_file(str(test_datadir / "s3_test_file"), "s3_test_file")
+    (
+        ebs_volume_id,
+        existing_efs_id,
+        existing_fsx_lustre_fs_id,
+        fsx_ontap_volume_id,
+        fsx_open_zfs_volume_id,
+    ) = _create_shared_storages_resources(
+        snapshots_factory,
+        request,
+        vpc_stack,
+        region,
+        efs_stack_factory,
+        fsx_factory,
+        svm_factory,
+        open_zfs_volume_factory,
+        bucket_name,
+    )
+
+    # Create cluster with initial configuration
+    init_config_file = pcluster_config_reader()
+    cluster = clusters_factory(init_config_file)
+    remote_command_executor = RemoteCommandExecutor(cluster)
+    scheduler_commands = scheduler_commands_factory(remote_command_executor)
+    cluster_nodes = scheduler_commands.get_compute_nodes()
+    queue1_nodes = [node for node in cluster_nodes if "queue1" in node]
+
+    # submit a job to queue1
+    queue1_job_id = scheduler_commands.submit_command_and_assert_job_accepted(
+        submit_command_args={
+            "command": "sleep 3000",
```

Comment:
At first, such a long sleep time sounds unreasonable, but then looking at the rest of the test it makes sense. I suggest to describe in a short comment why you need such a long sleep time.

Reply:
added